tests/integration/test_filter.py

```diff
@@ -157,20 +157,35 @@ def test_filter_year(c):
         (
             "SELECT * FROM parquet_ddf WHERE (b > 5 AND b < 10) OR a = 1",
             lambda x: x[((x["b"] > 5) & (x["b"] < 10)) | (x["a"] == 1)],
-            [[("a", "==", 1)], [("b", "<", 10), ("b", ">", 5)]],
+            [
+                [("a", "==", 1), ("b", "<", 10)],
+                [("a", "==", 1), ("b", ">", 5)],
+                [("b", ">", 5), ("b", "<", 10)],
+                [("a", "==", 1)],
+            ],
```

Comment:
The latest version of datafusion applies some conversions that convert filters to a cnf like format. So the dnf here 
`(b > 5 AND b < 10) OR a = 1` gets remapped to 
`(b>5 or a=1) and (b<10 or a=1)`  by datafusion which becomes this is dnf
`(b>5 and a=1) or (b<10 and a=1) or (b>5 and b<10) or (a=1)` .

I don't know if there's an easy way to simplify these predicate expressions but some cases might lead to larger number of redundant filters because of this.